extra/changelog-generator/changelog-generator

```diff
@@ -416,13 +462,16 @@ for repo in repos:
         else:
             service_header += " (%s)\n" % range
         print(service_header)
+    fixes = []
+    feats = []
+    others = []
     for entry in entry_list:
-        entry = "* " + entry
-        # Indent all non-empty lines.
-        entry = re.sub(r"\n([^\n])", r"\n  \1", entry)
-        print(entry)
-    if len(entry_list) > 0:
-        print()
+        (entry, bucket) = strip_type(entry, fixes, feats, others)
+        bucket.append(entry)
+
+    print_category_entries(fixes, "Bug Fixes")
+    print_category_entries(feats, "Features")
+    print_category_entries(others, "Other", fixes or feats)
```

Comment:
`fixes or feats` - This one was new to me. It's the union of the two lists?

Reply:
Expression is false if and only if both lists are empty.

Reply:
The idea is that if the lists are empty then we do not print the `##### Other` string. This is to ensure that the script is still compatible with the tests in `../tests`.

Reply:
Ah, I misread the arguments. Smart!